src/__tests__/theme-provider.js

```diff
@@ -109,3 +110,69 @@ test('does nothing when receive same theme via props', () => {
   wrapper.setProps({theme})
   expect(wrapper).toMatchSnapshotWithGlamor(`with theme prop of margin 2px`)
 })
+
+test('does not override user defined theme contexts', () => {
+  const Child = glamorous.div((props, context) => ({
+    fontSize: context.theme.fontSize,
+  }))
+  Child.contextTypes = {
+    theme: PropTypes.object,
+  }
+
+  class Parent extends React.Component {
```

Comment:
Ran into a little bit of trouble when injecting `context` with the `ThemeProvider`, who also defines a child context:

https://gist.github.com/ajwhite/96955b52c1f29962b4d5f1cf13ed6202

I'll do some research on this, unless you've ran into this and have some tips off the top of your head